src/sentry/sentry_metrics/querying/metadata/metrics.py

```diff
@@ -0,0 +1,103 @@
+from collections import defaultdict
+from collections.abc import Sequence
+from typing import Any
+
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.snuba.metrics import parse_mri
+from sentry.snuba.metrics.datasource import (
+    _build_metric_meta,
+    get_metrics_blocking_state_of_projects,
+)
+from sentry.snuba.metrics.utils import BlockedMetric, MetricMeta
+from sentry.snuba.metrics_layer.query import fetch_metric_mris
+
+
+def get_metrics_meta(
+    organization: Organization,
+    projects: Sequence[Project],
+    use_case_ids: Sequence[UseCaseID],
+) -> Sequence[MetricMeta]:
```

Comment:
Can we define our own type for `MetricMeta` which is isolated in this module? I would tie to this the proper serialization. We could do this in a follow up PR, not strictly needed rn, but I would like to remove any dependency we have on `datasource` and `metrics.utils`.

Reply:
Agreed, we can and should do that, but for now I'd like to keep compliance with the previous format and do that work in a separate PR.